analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ClassShouldNotBeAbstract.CSharp9.cs

```diff
@@ -58,17 +58,22 @@ public void flipSwitch()
         }
     }
 
-    public abstract record View // FN
+    public abstract record View // Noncompliant {{Convert this 'abstract' record to an interface.}}
     {
         public abstract string Content { get; }
     }
 
-    public abstract record View2 // Compliant, has abstract and non abstract members
+    public abstract record View2() // Compliant, has abstract and non abstract members
     {
         public abstract string Content { get; }
         public abstract string Content1 { get; }
         public string Content2 { get; }
     }
 
-    public abstract record Record(string X);
+    public abstract record Record(string X); // Noncompliant {{Convert this 'abstract' record to a concrete type with a protected constructor.}}
+
+    public abstract record Record2(string X) // Compliant, this record has a propery X which is concrete
```

Comment:
The comment about the concrete X property is a bit confusing, as the  case is Compliant due to the Content abstract property.

Reply:
In this case the X property is relevant, without it we would suggest to use an interface since all methods are abstract.

Reply:
Yes, indeed. I missed that fact. Thanks for pointing it out.